scripts/startup.sh

```diff
@@ -226,29 +228,252 @@ case ${options[$?]} in
+      BOOTpart=${options[$?]%|*}
+      set_option BOOTPART ${BOOTpart%|*}
+    ;;
+    esac
+
+    echo -e "\n${BOOTpart%|*} selected as EFI partition. Is that OK?"
+    options=("Yes" "No")
+    select_option $? 1 "${options[@]}"
+    case ${options[$?]} in
+    y|Y|yes|Yes|YES)
+      return ;;
+    n|N|no|NO|No)
+      diskpart
+      return
+    ;;
+    *)
+      diskpart
+      return
+    ;;
+    esac
+
+    echo -e "\nDo you want to format the selected EFI partition?"
+    options=("Yes" "No")
+    select_option $? 1 "${options[@]}"
+    case ${options[$?]} in
+    y|Y|yes|Yes|YES) set_option FORMATEFI "yes" ;;
+    n|N|no|NO|No) set_option FORMATEFI "no" ;;
+    *) set_option FORMATEFI "no" ;;
+    esac
+
+  elif [[ $(sudo sudo fdisk -l | grep -i '^Disklabel type') = "Disklabel type: gpt" ]]; then # Checking for GPT Disk Label on a Legacy BIOS (non UEFI) System
+    clear
+    echo -en "
+    ------------------------------------------------------------------------
+        GUID Partition Table (GPT) detected on a Legacy BIOS System.
+        If you had installed Windows or any other OS in the current
+        disk drive, a BIOS BOOT partition may already exist in it.
+        Identify it from the table below and select it. But, if it
+        doesnot exist, then create a partition of size 100 MB and set
+        its type to BIOS BOOT.
+    ------------------------------------------------------------------------
+
+    "
+
+    fdisk -l
+
+    echo -e "\n"
+    echo "Select the BIOS BOOT Partition (This partition should be about 100-500 MB in size and should have BIOS boot label.): "
+    options=("Create a BIOS BOOT partition using cfdisk" $(lsblk -n --output TYPE,KNAME,SIZE,MOUNTPOINT | awk '$1=="part"{print "/dev/"$2"|"$3"__________"$4}'))
+    select_option $? 1 "${options[@]}"
+    case $? in
+    0)
+      manualpart
+      diskpart
+      return
+    ;;
+    *)
+      BOOTpart=${options[$?]%|*}
+      set_option BOOTPART ${BOOTpart%|*}
+      set_option FORMATEFI "no"
+    ;;
+    esac
+
+    echo -e "\n${BOOTpart%|*} selected as BIOS BOOT partition. Is that OK?"
+    options=("Yes" "No")
+    select_option $? 1 "${options[@]}"
+
+    case ${options[$?]} in
+    y|Y|yes|Yes|YES)
+      return ;;
+    n|N|no|NO|No)
+      diskpart
+      return
+    ;;
+    *)
+      diskpart
+      return
+    ;;
+    esac
+  fi
+;;
+
+1)
+  clear
+  echo -ne "
+------------------------------------------------------------------------
+    WARNING: ARCH LINUX WILL BE INSTALLED IN THE SELECTED DISK DRIVE.
+    This will delete all the data present in it, so make sure that
+    you select the right disk drive that donot contain any useful files.
+------------------------------------------------------------------------
 
-drivessd
+"
+
+  lsblk
+
+  echo -e "\nSelect the disk to install on: "
+  options=($(lsblk -n --output TYPE,KNAME,SIZE | awk '$1=="disk"{print "/dev/"$2"|"$3}'))
+
+  select_option $? 1 "${options[@]}"
+  disk=${options[$?]%|*}
```

Comment:
change disk and DISK

Reply:
As I said, the original script used disk and DISK, so I stuck with the same.

Reply:
@HiDe-Techno-Tips  lowercase is local variable and UPPERCASE is global variable throughout the scripts.

Reply:
@khaneliman, I understand and that is what I meant to say. I used UPPERCASE variable name to store in setup.conf and the lowercase `disk` pointed out by @mfgbhatti is the local variable and doesn't have to be in UPPERCASE.